modeltranslation/tests/tests.py

```diff
@@ -463,6 +478,84 @@ def test_constructor(self):
+
+        obj.title_de = 'de 2'
+        obj.save(update_fields=['title'])
+        assert_db_record(obj, title='de 2', title_de='de 2', title_en=None)
+
+        obj.title = 'de 3'
+        obj.save(update_fields=['title_de'])
+        assert_db_record(obj, title='de 2', title_de='de 3', title_en=None)
+
+        obj.title_de = 'de 4'
+        obj.save(update_fields=['title_de'])
+        assert_db_record(obj, title='de 2', title_de='de 4', title_en=None)
+
+        obj.title = 'de 5'
+        obj.title_en = 'en 5'
+        obj.save(update_fields=['title', 'title_en'])
+        assert_db_record(obj, title='de 5', title_de='de 5', title_en='en 5')
+
+        obj.title_de = 'de 6'
+        obj.title_en = 'en 6'
+        obj.save(update_fields=['title_de', 'title_en'])
+        assert_db_record(obj, title='de 5', title_de='de 6', title_en='en 6')
+
+        obj.title_de = 'de 7'
+        obj.title_en = 'en 7'
+        obj.save(update_fields=['title', 'title_de', 'title_en'])
+        assert_db_record(obj, title='de 7', title_de='de 7', title_en='en 7')
+
+    def test_save_active_translation_field(self):
+        with override('en'):
+            obj = models.TestModel.objects.create(title='old')
+            obj.title = 'en 1'
+            obj.save(update_fields=['title'])
+            assert_db_record(obj, title='en 1', title_de=None, title_en='en 1')
+
+            obj.title_en = 'en 2'
+            obj.save(update_fields=['title'])
+            assert_db_record(obj, title='en 2', title_de=None, title_en='en 2')
+
+            obj.title = 'en 3'
+            obj.save(update_fields=['title_en'])
+            assert_db_record(obj, title='en 2', title_de=None, title_en='en 3')
+
+            obj.title_en = 'en 4'
+            obj.save(update_fields=['title_en'])
+            assert_db_record(obj, title='en 2', title_de=None, title_en='en 4')
+
+            obj.title = 'en 5'
+            obj.title_de = 'de 5'
+            obj.save(update_fields=['title', 'title_de'])
+            assert_db_record(obj, title='en 5', title_de='de 5', title_en='en 5')
+
+            obj.title_de = 'de 6'
+            obj.title_en = 'en 6'
+            obj.save(update_fields=['title_de', 'title_en'])
+            assert_db_record(obj, title='en 5', title_de='de 6', title_en='en 6')
+
+            obj.title_de = 'de 7'
+            obj.title_en = 'en 7'
+            obj.save(update_fields=['title', 'title_de', 'title_en'])
+            assert_db_record(obj, title='en 7', title_de='de 7', title_en='en 7')
+
+    def test_save_non_original_translation_field(self):
+        obj = models.TestModel.objects.create(title='old')
+
+        obj.title_en = 'en value'
+        obj.save(update_fields=['title'])
+        assert_db_record(obj, title='old', title_de='old', title_en=None)
+
+        obj.save(update_fields=['title_en'])
+        assert_db_record(obj, title='old', title_de='old', title_en='en value')
```

Comment:
i think this two can also be move to pytest.param.

Reply:
I left it as is for more readability